### PR TITLE
Add pylint

### LINKED ISSUE #10: Setup Automatic Code Linting
1. Setup a Github action to automatically perform python code linting with _pylint_.
2. Ensure existing codebase is compliant with _pylint_.
3. Update documentation to inform developers about code standards.

### PR DESCRIPTION
**Pull request by jaredb1011**
Never used github actions before, so I may be doing something wrong here, but the action seemed to be working properly on the branch.

I figured we'd go ahead and start with pylint.

### CODE REVIEW

**Comment by jaredb1011** [comment]
I'm going to go ahead and fix the issues as well, so converting to draft while WIP

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
```
Comment (Huskydog9988) [inline .github/workflows/pylint.yml:20]: i think in future python versions you need to specify a virtual env on linux, but some more senior python devs might need to correct me on this.

**Review by Huskydog9988 — APPROVED** [review]

File: .github/workflows/pylint.yml (hunk 1 above)
Comment (jaredb1011) [reply .github/workflows/pylint.yml:20]: I also am not too sure about this. Let's address this later on if it becomes an issue.

**Review by Joe-TheBro — APPROVED** [review]
lgtm

**Pull request merged**